client/extensions/woocommerce/app/products/product-form-additional-details-card.js

```diff
@@ -0,0 +1,197 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+import { find, debounce } from 'lodash';
+import Gridicon from 'gridicons';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FoldableCard from 'components/foldable-card';
+import FormLabel from 'components/forms/form-label';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import FormTextInput from 'components/forms/form-text-input';
+import TokenField from 'components/token-field';
+
+class ProductFormAdditionalDetailsCard extends React.Component {
+
+	state = {
+		attributeNames: {},
+	};
+
+	static propTypes = {
+		product: PropTypes.object.isRequired,
```

Comment:
Please consider enforcing the shape of product, at least the fields this component relies on.